test/pytest/test_token_authorization.py

```diff
@@ -0,0 +1,226 @@
+import json
+import os
+import shutil
+import subprocess
+import tempfile
+import time
+from pathlib import Path
+
+import pytest
+import requests
+import test_utils
+
+ROOT_DIR = os.path.join(tempfile.gettempdir(), "workspace")
+REPO_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../")
+data_file_kitten = os.path.join(REPO_ROOT, "test/pytest/test_data/kitten.jpg")
+config_file = os.path.join(REPO_ROOT, "test/resources/config_token.properties")
+
+
+# Set up token plugin
+def get_plugin_jar():
+    new_folder_path = os.path.join(ROOT_DIR, "plugins-path")
+    plugin_folder = os.path.join(REPO_ROOT, "plugins")
+    os.makedirs(new_folder_path, exist_ok=True)
+    os.chdir(plugin_folder)
+    subprocess.run(["./gradlew", "formatJava"])
+    result = subprocess.run(["./gradlew", "build"])
+    jar_path = os.path.join(plugin_folder, "endpoints/build/libs")
+    jar_file = [file for file in os.listdir(jar_path) if file.endswith(".jar")]
+    if jar_file:
+        shutil.move(
+            os.path.join(jar_path, jar_file[0]),
+            os.path.join(new_folder_path, jar_file[0]),
+        )
+    os.chdir(REPO_ROOT)
+    result = subprocess.run(
+        f"python ts_scripts/install_from_source",
+        shell=True,
+        capture_output=True,
+        text=True,
+    )
+
+
+# Parse json file and return key
+def read_key_file(type):
+    json_file_path = os.path.join(REPO_ROOT, "key_file.json")
+    with open(json_file_path) as json_file:
```

Comment:
The test only works when called from test/pytest. Better to make it independent from calling directory.